sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client.py

```diff
@@ -584,36 +583,39 @@ def upload_blob(
             client=self._client,
             **kwargs)
         if blob_type == BlobType.BlockBlob:
-            return upload_block_blob(**options)
+            return upload_block_blob(**options)  # type: ignore
         if blob_type == BlobType.PageBlob:
-            return upload_page_blob(**options)
-        return upload_append_blob(**options)
+            return upload_page_blob(**options)  # type: ignore
+        return upload_append_blob(**options)  # type: ignore
 
     @overload
     def download_blob(
```

Comment:
azure\storage\blob\_blob_client.py:592: error: Overloaded function signatures 1 and 2 overlap with incompatible return types  [misc]